app/scripts/views/mixins/signin-mixin.js

```diff
@@ -55,19 +56,35 @@ define(function (require, exports, module) {
     },
 
     onSignInSuccess: function (account) {
-      if (! account.get('verified')) {
+      if (account.get('verified') && account.get('challenge') === Constants.SIGNIN_EMAIL) {
```

Comment:
Oh, I didn't notice [`tokenVerified` in the spec](https://github.com/mozilla/fxa/pull/143/files#diff-a3580b834127f11689379869c2407b38R126) earlier. Is `tokenVerified` a new field in addition to `verified`?